### PR TITLE
add support for globally registered iosrtc plugin

### PR DESCRIPTION
**Pull request by calebboyd**
#71

### CODE REVIEW

**Comment by calebboyd** [comment]
I signed it!

File: adapter.js [hunk 1]
```diff
@@ -448,6 +448,24 @@ if (typeof window === 'undefined' || !window.navigator) {
...
+  //detected iOS version
+  webrtcDetectedVersion = parseInt(navigator.appVersion.match(/OS (\d+)_/)[1], 10);
+  //minimun suppored iOS version
+  webrtcMinimumVersion = 7;
+  //NOTE: assume user has waited until deviceready before loading adapter.js, to avoid side-effects.
+  //NOTE: assume window.cordova.plugins.iosrtc.registerGlobals() has been called.
+  getUserMedia = navigator.getUserMedia;
+
+  attachMediaStream = function (element, stream) {
+    element.src = URL.createObjectURL(stream)
```
Comment (KaptenJansson) [inline adapter.js]: Semi colon missing.

Tip: use grunt locally to catch these issues.

File: adapter.js [hunk 2]
```diff
@@ -448,6 +448,24 @@ if (typeof window === 'undefined' || !window.navigator) {
...
+  };
+  reattachMediaStream = function (to, from) {
+    to.src = from.src
```
Comment (KaptenJansson) [inline adapter.js]: Semi colon missing.

**Comment by KaptenJansson** [comment]
Hmm wonder why the comments did not go in the right place...

File: adapter.js [hunk 3]
```diff
@@ -454,6 +455,25 @@ if (typeof window === 'undefined' || !window.navigator) {
...
     to.srcObject = from.srcObject;
   };
+} else if (window.cordova &&
+    window.cordova.plugins && window.cordova.plugins.iosrtc) {
+  webrtcUtils.log('This appears to be Safari');
+
+  webrtcDetectedBrowser = 'safari';
+  //detected iOS version
+  webrtcDetectedVersion = parseInt(navigator.appVersion.match(/OS (\d+)_/)[1], 10);
+  //minimun suppored iOS version
```
Comment (KaptenJansson) [inline adapter.js]: `//minimun suppored iOS version` should be `// Minimum suppored iOS version.`

File: adapter.js [hunk 4]
```diff
@@ -454,6 +455,25 @@ if (typeof window === 'undefined' || !window.navigator) {
   reattachMediaStream = function(to, from) {
     to.srcObject = from.srcObject;
   };
+} else if (window.cordova &&
+    window.cordova.plugins && window.cordova.plugins.iosrtc) {
+  webrtcUtils.log('This appears to be Safari');
+
+  webrtcDetectedBrowser = 'safari';
+  //detected iOS version
```
Comment (KaptenJansson) [inline adapter.js]: `//detected iOS version` should be `// Detected iOS version.`

File: adapter.js [hunk 5]
```diff
@@ -454,6 +455,25 @@ if (typeof window === 'undefined' || !window.navigator) {
...
+  webrtcDetectedVersion = parseInt(navigator.appVersion.match(/OS (\d+)_/)[1], 10);
+  //minimun suppored iOS version
+  webrtcMinimumVersion = 7;
+  //NOTE: assume user has waited until deviceready before loading adapter.js, to avoid side-effects.
```
Comment (KaptenJansson) [inline adapter.js]: Add space after //

File: adapter.js [hunk 6]
```diff
@@ -454,6 +455,25 @@ if (typeof window === 'undefined' || !window.navigator) {
...
+  //NOTE: assume window.cordova.plugins.iosrtc.registerGlobals() has been called.
```
Comment (KaptenJansson) [inline adapter.js]: Add space after //

**Comment by calebboyd** [comment]
Fixed the comments, sorry about the slop.

File: adapter.js [hunk 7]
```diff
@@ -454,6 +455,25 @@ if (typeof window === 'undefined' || !window.navigator) {
...
     to.srcObject = from.srcObject;
   };
+} else if (window.cordova &&
+    window.cordova.plugins && window.cordova.plugins.iosrtc) {
+  webrtcUtils.log('This appears to be Safari');
+
+  webrtcDetectedBrowser = 'safari';
+  // Detected iOS version
+  webrtcDetectedVersion = parseInt(navigator.appVersion.match(/OS (\d+)_/)[1], 10);
+  // Minimum supported iOS version
```
Comment (KaptenJansson) [inline adapter.js]: Add punctuation

File: adapter.js [hunk 8]
```diff
@@ -454,6 +455,25 @@ if (typeof window === 'undefined' || !window.navigator) {
   reattachMediaStream = function(to, from) {
     to.srcObject = from.srcObject;
   };
+} else if (window.cordova &&
+    window.cordova.plugins && window.cordova.plugins.iosrtc) {
+  webrtcUtils.log('This appears to be Safari');
+
+  webrtcDetectedBrowser = 'safari';
+  // Detected iOS version
```
Comment (KaptenJansson) [inline adapter.js]: Add punctuation

**Comment by calebboyd** [comment]
Fixed comments

**Comment by KaptenJansson** [comment]
Great ;) Thanks for putting up my tediousness ;). I will leave it up to @alvestrand to take a final look at it.

**Comment by calebboyd** [comment]
@alvestrand, anything else I need to do for this?

**Comment by KaptenJansson** [comment]
I just realised you should add a test for this.

**Comment by calebboyd** [comment]
I'm not entirely sure how that could be accomplished...any pointers?

File: adapter.js [hunk 9]
```diff
@@ -471,12 +468,21 @@ if (typeof window === 'undefined' || !window.navigator) {
...
-  attachMediaStream = function(element, stream) {
-    element.srcObject = stream;
-  };
-  reattachMediaStream = function(to, from) {
-    to.srcObject = from.srcObject;
-  };
+  webrtcUtils.createMediaStreamMethods();
+} else if (window.cordova &&
+    window.cordova.plugins && window.cordova.plugins.iosrtc) {
+  webrtcUtils.log('This appears to be Safari');
```
Comment (alvestrand) [inline adapter.js]: Nit: This log message should be "Safari with IOSRTC plugin".
We shouldn't assume that this is the only way Safari will achieve WebRTC compatibility.

File: adapter.js [hunk 10]
```diff
@@ -471,12 +468,21 @@ if (typeof window === 'undefined' || !window.navigator) {
...
+
+  webrtcDetectedBrowser = 'safari';
+  // Detected iOS version.
+  webrtcDetectedVersion = parseInt(navigator.appVersion.match(/OS (\d+)_/)[1], 10);
+  // Minimum supported iOS version.
+  webrtcMinimumVersion = 7;
```
Comment (alvestrand) [inline adapter.js:480]: Suggest "safari+iosrtc" in webrtcDetectedBrowser.

**Comment by alvestrand** [comment]
Added comments about naming (if this is only for IOSRTC, call it that).
For tests: See the tests that jan-ivar added for verifying that FF37 will be correctly detected. A similar pattern should allow you to check that the code actually does what you expect it to do.
(If we ever get multi-platform testing, we'll revisit this.)

**Comment by calebboyd** [comment]
Hmm. Without the device I can see it should work by doing the following: 

```
//shim cordova.plugins.iosrtc object
navigator.getUserMedia = navigator.webkitGetUserMedia;
navigator.webkitGetUserMedia = null;
var oldAppVersion = navigator.appVersion
navigator.appVersion = 'ios appversion'
invalidateRequireCacheForAdapter();
m = require('./adapter');
//assert
//restore, invalidate and require again
```

Thoughts?

**Comment by KaptenJansson** [comment]
Take a look at [this](https://github.com/webrtc/adapter/pull/88/files#diff-c1129c8b045390789fa8ff62f2c6b4a9R28) for pointers on how to test this (as mentioned by @alvestrand).

**Comment by calebboyd** [comment]
closing in favor of #119

**Pull request closed without merging**